packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -874,25 +874,22 @@ private static List<PatternItem> patternFromPigeon(
     return pattern;
   }
 
-  private static Cap toCap(Object o, AssetManager assetManager, float density) {
-    final List<?> data = toList(o);
-    switch (toString(data.get(0))) {
-      case "buttCap":
+  private static Cap capFromPigeon(
+      Messages.PlatformCap cap, AssetManager assetManager, float density) {
+    switch (cap.getType()) {
+      case BUTT_CAP:
         return new ButtCap();
-      case "roundCap":
+      case ROUND_CAP:
         return new RoundCap();
-      case "squareCap":
+      case SQUARE_CAP:
         return new SquareCap();
-      case "customCap":
-        if (data.size() == 2) {
-          return new CustomCap(toBitmapDescriptor(data.get(1), assetManager, density));
-        } else {
-          return new CustomCap(
-              toBitmapDescriptor(data.get(1), assetManager, density), toFloat(data.get(2)));
-        }
-      default:
-        throw new IllegalArgumentException("Cannot interpret " + o + " as Cap");
+      case CUSTOM_CAP:
+        assert cap.getRefWidth() != null;
```

Comment:
https://stackoverflow.com/questions/2758224/what-are-assertions-in-java-and-when-should-they-be-used 
I don't think this assertion protects against a non null value or provides a warning to the caller. 

Which of the values (data1 or data2) was refWidth?

Reply:
It was `data.get(2)`. Platform interface's custom type `Cap` always has a `refWidth`, so in the event that somehow this gets a custom cap with no width, it will now throw an `IllegalArgumentException`, which the old method also could have thrown if conversion fails.